### ISSUE TITLE
Pass a tar archive rather than a directory?

### ISSUE THREAD

**Issue by zbeekman** [megastep/makeself#62]
I like to create source distributions with `git archive` which knows to exclude some files based on `.gitattributes`. Is there a way to pass the script an extant archive rather than having it create one for you from a directory?

**Comment by megastep** [comment]
Not exactly, however there is a way to append an existing archive (it could be a minimal one) using the `--append` flag.

**Comment by zbeekman** [comment]
So, let me see if I understand the course of action you are proposing:

 1. Make the self-extracting archive with a basically empty archive (Maybe I need to have the script that makeself will call after extraction in this portion of the archive?)
 2. Use the append option to add the "real" archive full of the data I want added.

**Comment by megastep** [comment]
Yeah, however the append option doesn't allow you to specify an actual archive, the files also come from the directory you specify. But this would allow you to break up the creation of the archive into chunks if this helps with what you're trying to do.

**Comment by zbeekman** [comment]
I see... it sounds like my best bet is to `git archive` followed by a `tar
-xvf ...` into a dedicated directory, followed then by a re-tar etc with
makeself.

Can I put in a feature request to have `makeself` work directly from an
existing archive? (Or alternatively understand .gitattributes, which would
be much harder)

**Comment by megastep** [comment]
Handling `.gitattributes` directly seems out of the scope of this project, if only because it's not the only kind of SCM files it would have to handle, and I don't want it to be that specialized. 

However, being able to handle tar archives directly sounds like a reasonable addition.

**Comment by zbeekman** [comment]
@megastep FYI I change the title to reflect that this is a feature request.

**Comment by rirl** [comment]
Any change in status on this?
I would also find this feature useful.

Perhaps I could create a fork and give it a whirl if someone else has not started it?
